### PR TITLE
Gomatic to check migration durations

### PR DESCRIPTION
**Pull request by doctoryes**
https://openedx.atlassian.net/browse/TE-1885

Calls the tubular script which checks the duration of a stage edxapp migration and alerts via email when threshold is crossed.

@edx/pipeline-team @cpennington Please review.

### CODE REVIEW

File: edxpipelines/pipelines/config/edxapp.yml [hunk 1]
```diff
@@ -5,6 +5,16 @@ edxapp_subapps:
 play_name: "edxapp"
 db_migration_user: "www-data"
 
+# Migration duration alerting on stage.
+migration_duration_threshold: 10
+alert_from_address: fill@me.in
```
Comment (doctoryes) [inline edxpipelines/pipelines/config/edxapp.yml]: The email address that goes here needs to be SES-verified:
http://docs.aws.amazon.com/ses/latest/DeveloperGuide/verify-email-addresses.html

**Comment by doctoryes** [comment]
Another PR is required here in terraform to enable email sending from go-agents FYI.

File: edxpipelines/pipelines/config/edxapp.yml (hunk 1 above)
Comment (cpennington) [inline edxpipelines/pipelines/config/edxapp.yml]: Don't we have emails we send other alert emails from?

File: edxpipelines/patterns/stages.py [hunk 2]
```diff
@@ -606,6 +610,10 @@ def generate_run_migrations(pipeline,
         application_user (str): Username to use while running the migrations
         application_name (str): Name of the application (e.g. edxapp, programs, etc...)
         application_path (str): path of the application installed on the target machine
+        check_duration (bool): Should the migration duration be checked against a threshold and alerted if over it?
+        duration_threshold (int): Threshold in seconds over which a migration duration will be alerted.
```
Comment (cpennington) [inline edxpipelines/patterns/stages.py]: What about just having the presence of the `duration_threshold` determine if the check is going to happen?

File: tox.ini [hunk 3]
```diff
@@ -4,6 +4,8 @@ skip_install = True
 skipsdist = True
 
 [testenv]
-deps = -rrequirements/test_requirements.txt
+deps =
+    -rrequirements/test_requirements.txt
+    -rrequirements.txt
```
Comment (cpennington) [inline tox.ini]: `test_requirements.txt` doesn't automatically include `requirements.txt`?

File: edxpipelines/pipelines/config/edxapp.yml (hunk 1 above)
Comment (doctoryes) [inline edxpipelines/pipelines/config/edxapp.yml]: I intend to find out. @feanil  - do we use a AWS `us-east-1` SES-verified From: email address anywhere else?

File: edxpipelines/patterns/stages.py (hunk 2 above)
Comment (doctoryes) [inline edxpipelines/patterns/stages.py]: Sure - less params is better.

File: tox.ini (hunk 3 above)
Comment (doctoryes) [inline tox.ini]: It does - but the `-r ../requirements.txt` in the `test_requirements.txt' was *not* working in my tox environment. It complained about no `enum` module until I made this change.

**Pull request merged**